static/js/popovers.js

```diff
@@ -499,10 +502,18 @@ export function toggle_actions_popover(element, id) {
         const should_display_uncollapse =
             !message.locally_echoed && !message.is_me_message && message.collapsed;
 
-        const should_display_edit_and_view_source =
-            message.content !== "<p>(deleted)</p>" ||
-            editability === message_edit.editability_types.FULL ||
+        const should_display_edit_and_move_message =
+            (message.content !== "<p>(deleted)</p>" ||
+                editability === message_edit.editability_types.FULL ||
+                editability === message_edit.editability_types.TOPIC_ONLY) &&
+            editability !== message_edit.editability_types.NO &&
+            editability !== message_edit.editability_types.NO_LONGER;
+        const should_display_view_source =
+            (message.content !== "<p>(deleted)</p>" &&
+                editability === message_edit.editability_types.NO) ||
+            editability === message_edit.editability_types.NO_LONGER ||
```

Comment:
I think we can simplify these conditions.
For edit message we should show the icon only when editiability is `FULL` or `TOPIC_ONLY` and it should not depend on content is "deleted" or not.
And for view source, we can show if editiing is not `FULL` and content is not "deleted" I think.
@alya thoughts?

Reply:
Sorry, to clarify, is this about the three-dot menu options, or icons in the message header bar?

Reply:
Three-dot menu options.

Reply:
> For edit message we should show the icon only when editiability is `FULL` or `TOPIC_ONLY` and it should not depend on content is "deleted" or not.

Yes, that sounds good.

Reply:
> And for view source, we can show if editiing is not FULL and content is not "deleted" I think.

Possibly? Or maybe just any time editiing is not FULL? @timabbott for your thoughts

Reply:
Just checked with @timabbott -- we can just allow viewing source when editing is not FULL.

Reply:
Do we have any conditions about "(deleted)" elsewhere?

Reply:
> Do we have any conditions about "(deleted)" elsewhere?

No.

And I forgot to mention it in the previous comment, we would also need to show the `Move message` option when only changing stream is allowed and not topic and content editing.

Reply:
Ok, great, let's not have any logic depending on "(deleted)", then.